src/components/CheckboxGroup/types.ts

```diff
@@ -0,0 +1,68 @@
+import React from "react";
+
+type StringOption = {
+  key: string;
+  label: string;
+};
+
+type NodeOption = {
+  key: string;
+  label: React.ReactNode;
+};
+
+export type CheckboxOption =
+  | StringOption
+  | NodeOption
+  | Readonly<StringOption>
+  | Readonly<NodeOption>;
```

Comment:
Whats the difference in having it as read only AND  just the type? Wouldn't one suffice?

Reply:
Nope one is not assignable to the other. If we only have `readonly` then we force all consumers to use `as const` which is fine for static properties but is very difficult to do for dynamic ones.